react/AddressForm.tsx

```diff
@@ -1,46 +1,162 @@
-    return address[fragment.name] != null ? (
+    const field = fields[fragment.name as keyof Fields]
+    const labelName = field && field.label ? field.label : null
+
+    if (ignoredFields.has(fragment.name)) return null
+    if (address[fragment.name] == null) return null
+    if (numberHasWithoutOption(labelName)) return <NumberOption showCheckbox />
+
+    const getFieldProps = (fragment: LineFragment) => {
+      const maxLength = field && field.maxLength ? field.maxLength : null
+      const autoComplete =
+        field && field.autoComplete ? field.autoComplete : null
+      const required = field && field.required ? field.required : null
+      const label = (
+        <FormattedMessage id={`place-components.label.${labelName}`} />
+      )
+      const onChange = (event: React.ChangeEvent) => {
+        if (
+          event.target instanceof HTMLInputElement ||
+          event.target instanceof HTMLSelectElement
+        ) {
+          setAddress({
+            ...address,
+            [fragment.name]: event.target.value,
+          })
+        }
+      }
+      const fieldRequired = {
+        errorMessage: (
+          <FormattedMessage id={`place-components.error.field-required`} />
+        ),
+      }
+      const options =
+        field && (field as OptionsField).options
+          ? (field as OptionsField).options
+          : null
+      const value = address[fragment.name]
+
+      return {
+        label,
+        value,
+        onChange,
+        options,
+        ...(maxLength && { maxLength }),
+        ...(autoComplete && { autoComplete }),
+        ...(required && address[fragment.name].length == 0 && fieldRequired),
+      }
+    }
+
+    return (
       <span key={fragment.name} className="w-25 dib mh3">
-        <Input
-          placeholder={fragment.name}
-          label={fragment.name}
-          value={address[fragment.name]}
-          onChange={(event: any) => {
-            const newAddress = {
-              ...address,
-              [fragment.name]: event.target.value,
-            }
-            setAddress(newAddress)
-          }}
-        />
+        {hasOptions(field as OptionsField) ? (
+          <Dropdown {...getFieldProps(fragment)} />
+        ) : (
+          <Input {...getFieldProps(fragment)} />
+        )}
       </span>
-    ) : null
+    )
   }
 
   const parseLine = (line: LineFragment[], index: number) => [
     ...line.map(parseLineFragment),
-    <br className={'line' + (index + 1) + '-delimiter'} key={index} />,
+    <br key={index} />,
   ]
 
+  const onEditButtonClick = () => {
+    setIgnoredFields(getSummaryFields(display['minimal']))
+    setEditing(true)
+  }
+
+  const displayMode: keyof Display = editing ? 'minimal' : 'compact'
+
   return (
     <div>
+      <PlaceDetails display={displayMode} />
+      {!editing && (
+        <ButtonPlain onClick={onEditButtonClick} title="edit">
+          <FormattedMessage id="place-components.label.edit" />
```

Comment:
you should use the return value from the `defineMessages` here

Reply:
I'm using `defineMessages` to handle dynamic messages, in this case, I think the current way suffices. Would it be more desirable to add this to the `defineMessages` object?

Reply:
sure, if you are already using `defineMessages` in this file I think it would be better if all messages for this file were defined there.